src/player.js

```diff
@@ -0,0 +1,54 @@
+import Logger from "./logger";
+
+const stepSize = 10;
+
+export default class Player {
+  constructor() {
+    this.log = new Logger();
+    this.boundKeydown = Player.keydownCallback.bind(this);
+    this.boundMousedown = Player.mousedownCallback.bind(this);
+  }
+
+  init() {
+    this.log.info("Starting BES Player");
+
+    document.addEventListener("keydown", this.boundKeydown);
+
+    let progressBar = document.querySelector(".progbar");
+    progressBar.style.cursor = "pointer";
+    progressBar.addEventListener("click", this.boundMousedown);
+  }
+
+  static keydownCallback(e) {
+    this.log.info("Keydown");
+    if (e.key == " " || e.key == "p")
+      document.querySelector("div.playbutton").click();
+
+    if (e.key == "ArrowUp") document.querySelector("div.prevbutton").click();
+
+    if (e.key == "ArrowDown") document.querySelector("div.nextbutton").click();
+
+    if (e.key == "ArrowRight") {
+      let audio = document.querySelector("audio");
+      audio.currentTime = audio.currentTime + stepSize;
+    }
+
+    if (e.key == "ArrowLeft") {
+      let audio = document.querySelector("audio");
+      audio.currentTime = audio.currentTime - stepSize;
+    }
+
+    if (e.target == document.body) e.preventDefault();
```

Comment:
A common pattern is to `e.preventDefault()` immediately at the top of the method to prevent any unexpected propagation. What events from `document.body` does this prevent? Are there similar events on other elements that should also be cancelled?

Reply:
@emcniece would it make sense to have this inside each if catch so that it's only the keys I am looking for that get prevented?

`e.target == document.body` was just suggested on the internet to prevent things like space bar from scrolling the page.

Reply:
That might work, and it might solve the `cmd+r` case too. It'll add some duplication in the method, there might be a refactor to something like:

```js
if(e.key in [" ","p","ArrowUp","ArrowDown"...]){
  e.preventDefault()
  switch(e.key){
    case " ":
    case "p":
      document.querySelector("div.playbutton").click();
      break;
    case "ArrowUp":
      ...
  }
}
```

Reply:
JS keypress events are contextual, as you know - by browser standards, when target is body spacebar will scroll, and when target is a media element spacebar will play/pause. You could take a similar approach and only process keybind events when `e.target` is the media player, but this severely limits the concept of tab-global key shortcuts. 

Jira is a good example of software that has tab-global key shortcuts. How often do you get frustrated with mis-focusing an element, hitting a key, and getting prompted with some unexpected popup window?

Reply:
@emcniece agree with enable/disable in the future (as well as changing the bindings)